[SQLLINE-331] Autocompletion for schema, table, column names.

snuyanzin opened the pull request:
The PR offers implementation of #331 .
Data for autocompletion is loaded by steps (columns names will be loaded after the first call of completion for columns of this table).
Sql identifier quotes (based on the current dialect) will be automatically added if there are spaces,
escaped identifiers, case sensitive or user already started with sql identifier. 
For keywords, functions, tables, schemas, columns there was also added description showing while autocompletion.
I tried to cover it with different tests within 3 dialects (especially including sql identifiers `"`, `` ` ``, `[]` and store upper and lowcases)

After that the old functionality for table/column autocompletion seems to be useless and could be removed as everything could be covered with the new one.

julianhyde commented:
Please remove the "." from the end of the commit message.

Can you add a javadoc comment for `Dialect.getExtraNameCharacters()`. (Yeah, I know the other methods don't have comments.)

Otherwise looks good.

snuyanzin commented:
Thank you for review
I removed "." from the end of commit messages and added comment for `Dialect#getExtraNameCharacters`

also there is a short demo (https://asciinema.org/a/259879?speed=2.5) regarding this PR which shows autocompletion for Calcite/MariaDB/MSSQLServer

snuyanzin commented:
Added back check for skipMeta option (fastconnect) as faced with velocity issue while playing with mat-calcite-plugin and with huge amount of classes (tables there in one schema).

snuyanzin commented:
Merged as https://github.com/julianhyde/sqlline/commit/32a915c0848ce3f2df8ef00aa69f2b8fc4706ee0
